src/http/authentication/cert_auth.h

```diff
@@ -124,31 +116,23 @@ namespace ccf
       const std::shared_ptr<enclave::RpcContext>& ctx,
       std::string& error_reason) override
     {
-      const auto caller_cert_pem =
-        crypto::cert_der_to_pem(ctx->session->caller_cert);
-
-      std::unique_ptr<NodeCertAuthnIdentity> identity = nullptr;
+      auto caller_public_key_der =
```

Comment:
Unrelated to this PR but fixed it anyway: since the node ID changes, we no longer need to `foreach` the nodes table to get the information for that node. We simply extract the public key from the cert, hash it, and use the hash as the key in the nodes table.